### PR TITLE
Added noWarn annotation and changelog

### PR DESCRIPTION
**Pull request by mbudiu-vmw**

### CODE REVIEW

File: p4-16/spec/P4-16-spec.mdk [hunk 1]
```diff
@@ -7556,6 +7571,7 @@ The following table shows all P4 reserved annotations.
 | `optional`    | parameter is optional                              | [#sec-optional-parameter-annotations] |
 | `tableonly`   | action cannot be a default_action                   | [#sec-table-action-anno] |
 | `deprecated`  | Construct has been deprecated                      | [#sec-deprecated-anno]   |
+| `noWarn`      | Has a string argument; inhibits compiler warnings  |                          |
```
Comment (jafingerhut) [inline p4-16/spec/P4-16-spec.mdk:7574]: Seems it might be nice to include cross-reference to the section, as is currently done for other annotations.

File: p4-16/spec/P4-16-spec.mdk [hunk 2]
```diff
@@ -7973,14 +7989,6 @@ parser generator tool.
...
 Several other constant terminals appear in these rules:
 ~ Begin P4Example
-- SHL is <<
-- LE is <=
-- GE is >=
-- NE is !=
-- EQ is ==
-- PP is ++
-- AND is &&
-- OR is ||
```
Comment (jafingerhut) [inline p4-16/spec/P4-16-spec.mdk:7983]: Are these changes intentional?  If so, I guess related to the "Fixed grammar for right shifts" change?

Comment (mbudiu-vmw) [reply p4-16/spec/P4-16-spec.mdk:7983]: Yes, these were removed from the grammar.

**Comment by mbudiu-vmw** [comment]
I think we can merge this.
